### PR TITLE
Some init code refactoring

### PR DESCRIPTION
**Pull request by kolyshkin**
This is a preparation for #3385, separated out for easier review.

This is all cosmetical -- code move and removal, variables renaming, all that jazz.

See individual commits for details.

### CODE REVIEW

**Comment by kolyshkin** [comment]
@thaJeztah PTAL (I hope this is consumable, changes may look big but mostly it's just moving and removing).

**Comment by kolyshkin** [comment]
@opencontainers/runc-maintainers can we please merge this one? It really blocks my further work on #3385 and subsequent improvements to libcontainer/README. If there's anything that does not make sense here, please let me know.

File: libcontainer/integration/init_test.go [hunk 1]
```diff
@@ -9,27 +10,26 @@ import (
...
 	if len(os.Args) < 2 || os.Args[1] != "init" {
 		return
 	}
+	// This is the golang entry point for runc init, executed
+	// before TestMain() but after libcontainer/nsenter's nsexec().
 	runtime.GOMAXPROCS(1)
 	runtime.LockOSThread()
 	if err := libcontainer.StartInitialization(); err != nil {
-		logrus.Fatal(err)
+		fmt.Fprintln(os.Stderr, err)
```
Comment (AkihiroSuda) [inline libcontainer/integration/init_test.go:25]: Could you add a comment line to explain why not to use logrus here

Comment (kolyshkin) [reply libcontainer/integration/init_test.go:26]: Done. I've also removed one commit from this PR.

**Review by AkihiroSuda — APPROVED** [review]

**Review by AkihiroSuda — APPROVED** [review]

**Review by mrunalp — APPROVED** [review]

**Pull request merged**